### ISSUE TITLE
Error when specifying the range of an index with a UnitRange

### ISSUE THREAD

**Issue by jgidi** [mcabbott/Tullio.jl#175]
Hi!

If we perform an operation with `@tullio` that needs to specify the range of an index manually, as follows:
```julia
using Tullio

N = 10
A = rand(5)

@tullio B[i, j] := A[i] (j in 1:N)
```
It errors with a message:
 `ERROR: MethodError: no method matching similar(::Vector{Float64}, ::Type{Float64}, ::Tuple{Base.OneTo{Int64}, UnitRange{Int64}})`

I have found two workarounds so far.
 1) Write the value of `N` explicitly (which is very limiting):
 ```julia
@tullio B[i, j] := A[i] (j in 1:10)
```

2) Use `Base.OneTo` instead of a `UnitRange`:
 ```julia
@tullio B[i, j] := A[i] (j in Base.OneTo(N))
```

This seems to be because `similar` does not accept `UnitRange`s as axes, and probably `Tullio` made some conversion in the first case when `N` was provided explicitly, but I don't know why the same conversion is not performed when the range is given as `1:N`. Also, I can confirm this happens at least for the julia versions `1.6.7` and `1.9.0`.

While the second workaround circumvents the problem completely, I think this behavior is odd and worth mentioning.

Thanks in advance!

**Comment by mcabbott** [comment]
Yes this really ought to work. It can see the `1` so it should be smart enough to make a `OneTo` for you. (In fact I thought there was already code for this, but must be wrong.)

**Comment by jgidi** [comment]
I can effectively reproduce your example, and I think I can see where the problem is now. It seems that the code for that functionality only gets available after loading `OffsetArrays`, even for a `UnitRange` starting at `1`.

After `using OffsetArrays` my first example works, but an OffsetArray is returned, which shouldn't be the case:

```julia
julia> using OffsetArrays

julia> @tullio B[i, j] := A[i] (j in 1:N)
5×10 OffsetArray(::Matrix{Float64}, 1:5, 1:10) with eltype Float64 with indices 1:5×1:10:
 0.897518   0.897518   0.897518   …  0.897518   0.897518   0.897518
 ...
```

**Comment by jgidi** [comment]
This may seem like an odd question, but do we really need to check that `r.args[3] isa Integer`? I just tested removing that check on L635 and everything seems to work great.
Obviously, if we pass a non-integer `N` it will fail,
```julia
julia> N = 5.5;

julia> @tullio B[i, j] := A[i] (j in 1:N)
ERROR: MethodError: no method matching Base.OneTo(::Float64)
...
```
but I think that error is pretty obvious to the user (at least more than it is now) and I don't think there would be side effects besides yielding the same error if a literal Float is passed as `1:5.5`.

If you like the idea I can propose this (very small) PR.
